db/migrate/20241014031810_add_deleted_at_to_course_achievements.rb

```diff
@@ -0,0 +1,6 @@
+class AddDeletedAtToCourseAchievements < ActiveRecord::Migration[7.0]
+  def change
+    add_column :course_achievements, :deleted_at, :datetime
```

Comment:
Why does Course Achievements need soft-delete for course-user deletion?

Reply:
Actually, I am not sure about the logic. So this is just proposal change.

Reply:
`Course::UserAchievement` is not the same as `Course::Achievement`

Reply:
I reverted this one.